src/dyn_subgrid/dynFATESLandUseChangeMod.F90

```diff
@@ -0,0 +1,194 @@
+       ! dynFatesLandUse_file = dyn_file_type(landuse_filename, YEAR_POSITION_START_OF_TIMESTEP)
+       dynFatesLandUse_file = dyn_file_type(landuse_filename, YEAR_POSITION_END_OF_TIMESTEP)
+
+       ! Get initial land use data
+       num_points = (bounds%endg - bounds%begg + 1)
+       landuse_shape(1) = num_points ! Does this need an explicit array shape to be passed to the constructor?
+       do varnum = 1, num_landuse_transition_vars
+          landuse_transition_vars(varnum) = dyn_var_time_uninterp_type( &
+               dyn_file=dynFatesLandUse_file, varname=landuse_transition_varnames(varnum), &
+               dim1name=grlnd, conversion_factor=1.0_r8, &
+               do_check_sums_equal_1=.false., data_shape=landuse_shape)
+       end do
+       do varnum = 1, num_landuse_state_vars
+          landuse_state_vars(varnum) = dyn_var_time_uninterp_type( &
+               dyn_file=dynFatesLandUse_file, varname=landuse_state_varnames(varnum), &
+               dim1name=grlnd, conversion_factor=1.0_r8, &
+               do_check_sums_equal_1=.false., data_shape=landuse_shape)
+       end do
+    end if
+
+    ! Since fates needs state data during initialization, make sure to call
+    ! the interpolation routine at the start
+    call dynFatesLandUseInterp(bounds,init_state=.true.)
+
+  end subroutine dynFatesLandUseInit
+
+
+  !-----------------------------------------------------------------------
+  subroutine dynFatesLandUseInterp(bounds, init_state)
+
+    use dynTimeInfoMod , only : time_info_type
+    use clm_varctl     , only : use_cn
+
+    ! !ARGUMENTS:
+    type(bounds_type), intent(in) :: bounds       ! proc-level bounds
+    logical, optional, intent(in) :: init_state   ! fates needs state for initialization
+
+    ! !LOCAL VARIABLES:
+    integer                     :: varnum
+    integer                     :: i
+    logical                     :: init_flag
+    real(r8), allocatable       :: this_data(:)
+    character(len=*), parameter :: subname = 'dynFatesLandUseInterp'
+    !-----------------------------------------------------------------------
+    SHR_ASSERT_ALL(bounds%level == BOUNDS_LEVEL_PROC, subname // ': argument must be PROC-level bounds')
+
+    ! This shouldn't be called by cn currently, but return if it is
+    if (use_cn) return ! Use this as a protection in lieu of build namelist check?
+
+    init_flag = .false.
+    if (present(init_state)) then
+       init_flag = init_state
+    end if
+
+    ! Get the current year
+    call dynFatesLandUse_file%time_info%set_current_year()
+
+    if (dynFatesLandUse_file%time_info%is_before_time_series() .and. .not.(init_flag)) then
+       ! Reset the land use transitions to zero for safety
+       landuse_transitions(1:num_landuse_transition_vars,bounds%begg:bounds%endg) = 0._r8
+       landuse_states(1:num_landuse_state_vars,bounds%begg:bounds%endg) = 0._r8
+    else
+       ! Right now we don't account for the topounits
```

Comment:
I'm not sure what you mean by topounits?

Reply:
This is a copy/paste holdover from e3sm.  Addressed via 87b149f86fe87a0e07ef627d64febb304306fd7a